### ISSUE TITLE
Auto generated Resource.designer.cs for monoandroid target not using global namespace

### ISSUE THREAD

**Issue by Doraku** [novotnyllc/MSBuildSdkExtras#188]
Hello, I am not sure where to look to fix this issue or even if this is an issue with MSBuildSdkExtras or some internals of visual studio but when building a dll with a target on monoandroid, a Resource.designer.cs file is auto generated in the build process (\obj\...) and the attribute `System.CodeDom.Compiler.GeneratedCodeAttribute("Xamarin.Android.Build.Tasks", "1.0.0.0")` is not using the `global::` prefix, since I have a `System` namespace in my project the compiler complain about this. An easy work around is to rename my namespace obviously but I still think this is something we shouldn't have to do.
Also this I am such a genius at naming things, I have a Resource namespace. This does not bother net and netcoreapp targets but again the monoandroid one complain because it can't declare this Resource class. Is it even required? (and I can rename it in the main time)

This is my csproj:
```
<Project Sdk="MSBuild.Sdk.Extras/2.0.54">
  <PropertyGroup Label="Compilation">
    <LangVersion>latest</LangVersion>
    <TargetFrameworks>
      net48;
      netcoreapp3.0;
      monoandroid90;
    </TargetFrameworks>
    <Optimize>true</Optimize>
    <Configurations>Debug;Release</Configurations>
  </PropertyGroup>

  <PropertyGroup Label="Debug" Condition="'$(Configuration)'=='Debug'">
    <Optimize>false</Optimize>
    <DefineConstants>DEBUG</DefineConstants>
  </PropertyGroup>
...
</Project>
```

This is the Resource.designer.cs file auto generated:
```
#pragma warning disable 1591
//------------------------------------------------------------------------------
// <auto-generated>
//     This code was generated by a tool.
//     Runtime Version:4.0.30319.42000
//
//     Changes to this file may cause incorrect behavior and will be lost if
//     the code is regenerated.
// </auto-generated>
//------------------------------------------------------------------------------

[assembly: global::Android.Runtime.ResourceDesignerAttribute("Talio.Resource", IsApplication=false)]

namespace Talio
{
	
	
	[System.CodeDom.Compiler.GeneratedCodeAttribute("Xamarin.Android.Build.Tasks", "1.0.0.0")]
	public partial class Resource
	{
		
		static Resource()
		{
			global::Android.Runtime.ResourceIdManager.UpdateIdValues();
		}
		
		public partial class Attribute
		{
			
			static Attribute()
			{
				global::Android.Runtime.ResourceIdManager.UpdateIdValues();
			}
			
			private Attribute()
			{
			}
		}
	}
}
#pragma warning restore 1591

```

I am using Visual Studio Community 2019 v16.3.2 and MSBuild.Sdk.Extras v2.0.54 as stated in the csproj. A quick repro would be to add a types in the `<project>.Resource` and `<project>.System` namespaces in a project targeting monoandroid.

**Comment by onovotny** [comment]
This sounds like an issue with the the underlying Xamarin tools. I'd suggest filing an issue here: https://github.com/xamarin/xamarin-android

**Comment by Doraku** [comment]
xamarin/xamarin-android#3744

Thanks, I'll keep this issue updated if something comes up over there.

**Comment by Doraku** [comment]
Not sure how to proceed but this issue has been closed on their side so once a release with the fix is made available I guess there will be something somewhere to update in MSBuildSdkExtras to take advantage of it :)

**Comment by clairernovotny** [comment]
Glad to see that it's fixed. It's not clear that there's any updates needed on the Extras for this, just using the VS version that contains the fix.

**Comment by Doraku** [comment]
Oh great, I will test this on future vs updates to close this issue once it is in then 👍

**Comment by Doraku** [comment]
Forgot to close this but the System namespace issue has been resolved at least since VS version 16.7.7 (sorry can't be sure didn't got the chance to work on my projects for some times). The automatically generated Resource class is still a problem but I guess they don't intend to change that behavior :/
